osquery/sql/sqlite_string.cpp

```diff
@@ -210,6 +210,76 @@ static void regexStringMatchFunc(sqlite3_context* context,
                       SQLITE_TRANSIENT);
 }
 
+/**
+ * @brief Concatenate strings ignoring nulls
+ */
+static void concatStringFunc(sqlite3_context* context,
+                             int argc,
+                             sqlite3_value** argv) {
+  // Nothing to concat, no point.
+  if (argc == 0) {
+    return;
+  }
+
+  std::string output;
+
+  for (auto i = 0; i < argc; i++) {
+    if (SQLITE_NULL == sqlite3_value_type(argv[i])) {
+      continue;
+    }
+
+    output.append(reinterpret_cast<const char*>(sqlite3_value_text(argv[i])));
+  }
+
+  sqlite3_result_text(context,
+                      output.c_str(),
+                      static_cast<int>(output.size()),
```

Comment:
Casting a `size_t` (unsigned integer) to a signed `int` could cause integer underflow/overflow.

If it must be int, one way to make it safe would be:

```c++
if (data > std::numeric_limits<int>::max())
        throw std::exception("Invalid cast.");
    return std::static_cast<int>(data);
```

Reply:
Hrm. if we think this is an anti-pattern, we have it in several places. I can add the guard you're suggesting pretty easily.

Reply:
I'll check with the team and get back to you, because I wouldn't want to add unnecessary checks in a bunch of places. Maybe this is the job for one of those `tryto` utility functions we have.

Reply:
Second opinion was, yea probably the `numeric_limits` approach is fastest. Thanks for tossing that check on there. Changing it elsewhere is probably a low priority for now.